### DISCUSSION TITLE
Is the termination condition of the experiment not working? For example termination="80% capacity"

### DISCUSSION THREAD

**Discussion by RobinBluehub** [pybamm-team/PyBaMM#2329] [Q&A]
I am experimenting with the SOC interval and found that the capacity termination condition is invalid, and the capacity degradation curve cannot be directly drawn when drawing the results. In my memory, it used to be possible.

'''KeyError: "'Capacity [A.h]' not found. Best matches are ['Measured capacity [A.h]', 'Throughput capacity [A.h]', 'Loss of capacity to SEI [A.h]']"'''

**Comment by valentinsulzer** [comment]
Please post full code

**Reply by RobinBluehub** [reply]
`import pybamm
import numpy as np
import matplotlib.pyplot as plt
pybamm.set_logging_level("NOTICE")
from pybamm import Parameter, constants, exp
options={
        "particle": "Fickian diffusion",
        "SEI":"solvent-diffusion limited",
        "particle mechanics":("swelling and cracking","swelling only"),
        "SEI porosity change":"true",
        "loss of active material":"stress-driven",
        "lithium plating": "partially reversible",
        "SEI on cracks":"true",
        #"thermal":"lumped",
}
def graphite_cracking_rate_Ai2020(T_dim):
    k_cr = 3.9e-20
    Eac_cr = 0  # to be implemented
    arrhenius = exp(Eac_cr / constants.R * (1 / T_dim - 1 / 298.15))
    return k_cr * arrhenius

def cracking_rate_Ai2020(T_dim):
    k_cr = 3.9e-20
    Eac_cr = 0  # to be implemented
    arrhenius = exp(Eac_cr / constants.R * (1 / T_dim - 1 / 298.15))
    return k_cr * arrhenius

model=pybamm.lithium_ion.DFN(options=options)
param1=pybamm.ParameterValues("OKane2022")
#SEI
param1.update({"Outer SEI solvent diffusivity [m2.s-1]":2.5E-21})
param1.update({"Initial inner SEI thickness [m]":2.5e-09})
param1.update({"Inner SEI reaction proportion":0.5})
#craking
param1.update({"Negative electrode cracking rate":graphite_cracking_rate_Ai2020})
param1.update({"Positive electrode cracking rate":cracking_rate_Ai2020})
#lam
param1.update({"Negative electrode LAM constant proportional term [s-1]": 2.78e-8})
param1.update({"Positive electrode LAM constant proportional term [s-1]": 2.78e-8})
#li_plating
param1.update({"Dead lithium decay constant [s-1]":1E-4})
param1.update({"Lithium plating kinetic rate constant [m.s-1]": 1E-9})

var_pts={"x_n": 30, "x_s":25, "x_p": 30, "r_n": 30, "r_p":30}
solver=pybamm.CasadiSolver(mode="safe", dt_max=200,max_step_decrease_count=20)

experiment_standardize = pybamm.Experiment(
    [(
        "Hold at 4.2V until 50mA ",
        "Rest for 2 hours ",
        "Discharge at 0.3C until 2.5V ",
        "Rest for 1 hours",
        "Charge at 0.3C until 4.2V ",
        "Hold at 4.2V until 50mA ",
        "Discharge at 1C for 0.5 hours"
    )],period="20 seconds"
)
experiment=pybamm.Experiment(
    [(
        "Charge at 0.5C for 1 hours or until 4.2V",
        "Rest for 0.5 hours",
        "Discharge at 1C for 0.5 hours",
        "Rest for 0.5 hours",
        )]*200,termination="80% capacity"
)
sim_standardize=pybamm.Simulation(model=model,parameter_values=param1,experiment=experiment_standardize,var_pts=var_pts,solver=solver)
solution_standardize1=sim_standardize.solve(calc_esoh=False)

sim=pybamm.Simulation(model=model,parameter_values=param1,experiment=experiment,var_pts=var_pts,solver=solver)
sol=sim.solve(calc_esoh=False)
`
KeyError: "'Capacity [A.h]' not found. Best matches are ['Measured capacity [A.h]', 'Throughput capacity [A.h]', 'Loss of capacity to SEI [A.h]']"

**Reply by RobinBluehub** [reply]
Also I would like to ask, on the basis of the above experiments, after adding the lumped temperature model and updating the temperature-related parameters in ORegan2022, the running time of a single cycle (CCCV, CC) is up to 10 minutes. Is there any way to optimize the time?
Thanks for your reply and good luck with your research！

**Reply by valentinsulzer** [reply]
If `calc_esoh=False` then you can't do the termination.
You can try to change mesh size and solver parameters to optimize the time.

**Reply by RobinBluehub** [reply]
First of all, thank you for taking the time to reply to me. However, when calc_esoh=True, the program will report an error. I don’t know if it is the reason why I set the result of the first experiment as the start of the second experiment. 
The error is as follows: pybamm.expression_tree.exceptions.SolverError: Could not solve for summary variables, run `sim.solve(calc_esoh=False)` to skip this step

**Reply by ch662** [reply]
@RobinBluehub I have been having the same issue. Did you ever find a fix?

**Reply by RobinBluehub** [reply]
sim.solve(calc_esoh=True)

**Reply by ch662** [reply]
fair enough, lol. Thank you for the quick reply. You saved me a decent amount of time.
